os/src/ZipOps.scala

```diff
@@ -255,6 +289,29 @@ object unzip {
       includePatterns: Seq[Regex] = List()
   ): os.Path = {
     stream(os.read.stream(source), dest, excludePatterns, includePatterns)
+    // preserve permissions
+    if (!scala.util.Properties.isWin) {
+      val zipFS = FileSystems.newFileSystem(
+        new URI("jar", source.wrapped.toUri.toString, null),
+        Map(
+          "create" -> "true",
+          "enablePosixFileAttributes" -> "true",
+          "defaultPermissions" -> Set.empty[String].asJava
+        ).asJava
+      )
+      try {
+        os.walk(dest).foreach { path =>
+          val relPath = path.subRelativeTo(dest).toString
+          if (os.zip.shouldInclude(relPath, excludePatterns, includePatterns)) {
+            val entry = zipFS.getPath(relPath)
+            val permissions = Files.getPosixFilePermissions(entry)
```

Comment:
I think you should use the more low-level `Files.getAttribute(entry, "zip:permissions")` here. 
In case the zip entry does not have permission information, `Files.getAttribute(entry, "zip:permissions")` would be empty and we could skip `os.perms.set`.

If using `Files.getPosixFilePermissions`, according to the doc

> The "permissions" attribute is not optional in the "posix" view so a default set of permissions are used for entries that do not have access permissions stored in the Zip file. The default set of permissions are
  OWNER_READ
  OWNER_WRITE
  GROUP_READ

Reply:
But the unzipped file would get some default permissions even if we dont set them, right?
With this approach we at least can set it with the "defaultPermissions" property.
Currently I set it to empty

Reply:
Yeah, you are right, I'm not super familiar with *nix permissions.  
Fixed now, looks good?

Reply:
So I didn't have a correct understanding of umask either. There're no default permissions in Unix. Applications set permissions to files they create. Umask will then mask some of the permissions. What you did before was totally fine. https://askubuntu.com/questions/44542/what-is-umask-and-how-does-it-work